flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerImplTest.java

```diff
@@ -1301,16 +1305,15 @@ private SlotRequest createSlotRequest(JobID jobId, ResourceProfile resourceProfi
 	}
 
 	private SlotManagerImpl createSlotManager(ResourceManagerId resourceManagerId, ResourceActions resourceManagerActions) {
-		return createSlotManager(resourceManagerId, resourceManagerActions, 1);
+		return createSlotManager(resourceManagerId, resourceManagerActions, new Configuration());
 	}
 
-	private SlotManagerImpl createSlotManager(ResourceManagerId resourceManagerId, ResourceActions resourceManagerActions, int numSlotsPerWorker) {
-		SlotManagerImpl slotManager = SlotManagerBuilder.newBuilder()
+	private SlotManagerImpl createSlotManager(ResourceManagerId resourceManagerId, ResourceActions resourceManagerActions, Configuration configuration) {
+		return SlotManagerBuilder.newBuilder()
 			.setDefaultWorkerResourceSpec(WORKER_RESOURCE_SPEC)
-			.setNumSlotsPerWorker(numSlotsPerWorker)
-			.build();
-		slotManager.start(resourceManagerId, Executors.directExecutor(), resourceManagerActions);
-		return slotManager;
+			.setNumSlotsPerWorker(configuration.getInteger(TaskManagerOptions.NUM_TASK_SLOTS))
+			.setMaxSlotNum(configuration.getInteger(ResourceManagerOptions.MAX_SLOT_NUM))
+			.buildWithDirectExec(resourceManagerId, Executors.directExecutor(), resourceManagerActions);
```

Comment:
```suggestion
			.buildAndStartWithDirectExec(resourceManagerId, Executors.directExecutor(), resourceManagerActions);
```
`buildWithDirectExec` already has `WithDirectExec` in its name so I would not pass `Executors.directExecutor()` to it but already have it inside `buildWithDirectExec`.